packages/node/src/dev-server.ts

```diff
@@ -82,7 +92,141 @@ function listen(server: Server, port: number, host: string): Promise<void> {
   });
 }
 
-let bridge: Bridge | undefined = undefined;
+async function createServerlessEventHandler(
+  entrypoint: string,
+  options: { shouldAddHelpers: boolean }
+): Promise<(request: IncomingMessage) => Promise<VercelProxyResponse>> {
+  const launcher = getVercelLauncher({
+    entrypointPath: entrypoint,
+    helpersPath: './helpers.js',
+    shouldAddHelpers: options.shouldAddHelpers,
+    useRequire,
+
+    // not used
+    bridgePath: '',
+    sourcemapSupportPath: '',
+  });
+  const bridge: Bridge = launcher();
+
+  return async function (request: IncomingMessage) {
+    const body = await rawBody(request);
+    const event = {
+      Action: 'Invoke',
+      body: JSON.stringify({
+        method: request.method,
+        path: request.url,
+        headers: request.headers,
+        encoding: 'base64',
+        body: body.toString('base64'),
+      }),
+    };
+
+    return bridge.launcher(event, {
+      callbackWaitsForEmptyEventLoop: false,
+    });
+  };
+}
+
+async function serializeRequest(message: IncomingMessage) {
+  const bodyBuffer = await streamToBuffer(message);
+  const body = bodyBuffer.toString('base64');
+  return JSON.stringify({
+    url: message.url,
+    method: message.method,
+    headers: message.headers,
+    body,
+  });
+}
+
+async function createEdgeEventHandler(
+  entrypoint: string
+): Promise<(request: IncomingMessage) => Promise<VercelProxyResponse>> {
+  const buildResult = await ncc(entrypoint);
+  const edgeFunctionDefinition = buildResult.code;
+
+  const initialCode = `
+    ${edgeFunctionDefinition};
+
+    addEventListener('fetch', async (event) => {
+      let serializedRequest = await event.request.text();
+      let requestDetails = JSON.parse(serializedRequest);
+
+      let body;
+
+      if (requestDetails.method !== 'GET' && requestDetails.method !== 'HEAD') {
+        body = Uint8Array.from(atob(requestDetails.body), c => c.charCodeAt(0));
+      }
+
+      let requestUrl = requestDetails.headers['x-forwarded-proto'] + '://' + requestDetails.headers['x-forwarded-host'] + requestDetails.url;
+
+      let request = new Request(requestUrl, {
+        headers: requestDetails.headers,
+        method: requestDetails.method,
+        body: body
+      });
+
+      event.request = request;
+
+      let edgeHandler = module.exports.default;
+      let response = edgeHandler(event.request, event);
+      return event.respondWith(response);
+    })`;
+
+  const edgeRuntime = new EdgeRuntime({
+    initialCode,
+    extend: (context: Primitives) => {
+      Object.assign(context, {
+        __dirname: '',
+        module: {
+          exports: {},
+        },
```

Comment:
These additions to the runtime context were required to get the compiled output of `ncc` to run. Does this look like a complete list of everything we made need?

Reply:
Take a look at the Next.js implementation here https://github.com/vercel/next.js/blob/88d0440ad4adf5c9ad6d9bf93babe9c48a9607f3/packages/next/server/web/sandbox/context.ts#L102